lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1862,15 +1862,19 @@ constexpr ExecutionContextScope *g_no_exe_ctx = nullptr;
     if ((TYPE) && !ReconstructType(TYPE))                                      \
       return result;                                                           \
     ExecutionContext _exe_ctx(EXE_CTX);                                        \
-    auto swift_scratch_ctx_lock = SwiftScratchContextLock(                         \
+    auto swift_scratch_ctx_lock = SwiftScratchContextLock(                     \
         _exe_ctx == ExecutionContext() ? nullptr : &_exe_ctx);                 \
     bool equivalent =                                                          \
         !ReconstructType(TYPE) /* missing .swiftmodule */ ||                   \
         (Equivalent(result, GetSwiftASTContext()->REFERENCE ARGS));            \
     if (!equivalent)                                                           \
       llvm::dbgs() << "failing type was " << (const char *)TYPE << "\n";       \
-    assert(equivalent &&                                                       \
-           "TypeSystemSwiftTypeRef diverges from SwiftASTContext");            \
+    /* Only assert if it's not an imported type. The typeref type system and   \
+    the ast context often have very different views of the same type           \
+     without there being anything nece wrong. */                               \
+    if (!IsImportedType(TYPE, nullptr))                                        \
+      assert(equivalent &&                                                     \
+             "TypeSystemSwiftTypeRef diverges from SwiftASTContext");          \
```

Comment:
I think this is fine if temporary, but should it be permanent? Do we know that this doesn't silence some real issues?

Reply:
We still have the llvm::dbgs right before this. If you think it's too broad we can add the failing type to `ShouldSkipValidation` instead.

Reply:
I am worried we wouldn't see the debug output. Also, our tests are far from exhaustive.

A higher level question is how long we plan to keep the assertions. My understanding is they were added to ensure the transition to TypeSystemSwiftTypeRef. Maybe the time will soon come that we delete these assertions, in which case maybe this is fine. However we could argue that we'd be less confident to remove the assertions given that we've weakened this one without fully knowing.

Reply:
Ok here's a patch skipping validation for this specific type https://github.com/apple/llvm-project/pull/4915